src/tmt_web/api.py

```diff
@@ -0,0 +1,163 @@
+
+from tmt_web import service
+from tmt_web.generators import html_generator
+
+app = FastAPI()
+
+
+class TaskOut(BaseModel):
+    id: str
+    status: str
+    result: str | None = None
+    status_callback_url: str | None = None
+
+
+# Sample url: https://tmt.org/?test-url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke
+# or for plans: https://tmt.org/?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic
+@app.get("/")
+def find_test(
+        test_url: Annotated[
+            str | None,
+            Query(
+                alias="test-url",
+                title="Test URL",
+                description="URL of a Git repository containing test metadata",
+            ),
+        ] = None,
+        test_name: Annotated[
+            str | None,
+            Query(
+                alias="test-name",
+                title="Test name",
+                description="Name of the test",
+            ),
+        ] = None,
+        test_ref: Annotated[
+            str | None,
+            Query(
+                alias="test-ref",
+                title="Test reference",
+                description="Reference of the test repository",
+            ),
+        ] = None,
+        test_path: Annotated[
+            str | None,
+            Query(
+                alias="test-path",
+                title="Test path",
+                description="Path to the test metadata directory",
+            ),
+        ] = None,
+        plan_url: Annotated[
+            str | None,
+            Query(
+                alias="plan-url",
+                title="Plan URL",
+                description="URL of a Git repository containing plan metadata",
+            ),
+        ] = None,
+        plan_name: Annotated[
+            str | None,
+            Query(
+                alias="plan-name",
+                title="Plan name",
+                description="Name of the plan",
+            ),
+        ] = None,
+        plan_ref: Annotated[
+            str | None,
+            Query(
+                alias="plan-ref",
+                title="Plan reference",
+                description="Reference of the plan repository",
+            ),
+        ] = None,
+        plan_path: Annotated[
+            str | None,
+            Query(
+                alias="plan-path",
+                title="Plan path",
+                description="Path to the plan metadata directory",
+            ),
+        ] = None,
+        out_format: Annotated[Literal["html", "json", "yaml"], Query(alias="format")] = "json",
+) -> TaskOut | str | Any:
+    # Parameter validations
+    if (test_url is None and test_name is not None) or (test_url is not None and test_name is None):
+        return "Invalid arguments!"
+    if (plan_url is None and plan_name is not None) or (plan_url is not None and plan_name is None):
+        return "Invalid arguments!"
+    if plan_url is None and plan_name is None and test_url is None and test_name is None:
```

Comment:
What about something like:

```
if all(arg is None for arg in [plan_url, plan_name, test_url, test_name]):
    return "Missing arguments!"
```

Reply:
@seberm let's not block on this and create a separate PRs for codestyle, ruff, etc?